element-{web,desktop}: 1.11.91 -> 1.11.95

transcaffeine opened the pull request:
## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).


---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

transcaffeine committed:
element-web: 1.11.91 -> 1.11.95

Release notes:
   - https://github.com/element-hq/element-web/releases/tag/v1.11.92
   - https://github.com/element-hq/element-web/releases/tag/v1.11.93
   - https://github.com/element-hq/element-web/releases/tag/v1.11.94
   - https://github.com/element-hq/element-web/releases/tag/v1.11.95
Full Changelog: https://github.com/element-hq/element-web/compare/v1.11.91...v1.11.95

Notable changes:
  - Device dehydration support on 'Set up recovery key' page
  - Render invite rejection reasons

transcaffeine committed:
element-desktop: 1.11.91 -> 1.11.95

Release notes: see element web
Full Changelog: https://github.com/element-hq/element-desktop/compare/v1.11.91...v1.11.95

NickCao commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 381196`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>element-desktop</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>element-web (element-web-unwrapped)</li>
  </ul>
</details>

teutat3s commented:
Waiting for https://nixpk.gs/pr-tracker.html?pr=380954 to reach the binary cache, can't build `electron` + `element-desktop` right now.

teutat3s commented:
Looks like it wants to use `git` for something now:
```
yarn install v1.22.22
[1/5] Validating package.json...
[2/5] Resolving packages...
warning Resolution field "@types/node@18.19.71" is incompatible with requested version "@types/node@^17.0.36"
error Couldn't find the binary git
```

transcaffeine commented:
With `element-desktop` having updated to a `node-gyp` version in it's `yarn.lock` which resolves to a github URL, this seems to be currently blocked until `fixup-yarn-lock` can handle this case.
See https://raw.githubusercontent.com/element-hq/element-desktop/refs/heads/develop/yarn.lock `@electron/node-gyp` -> `resolved`

transcaffeine commented:
While this does build and works (atleast from what i can tell), i am very unhappy about not solving the "`yarn install --offline` ignores the tarball from `fetchYarnDeps`"... This is honestly not a pretty workaround :/

transcaffeine commented:
On `aarch64-darwin`, some nodejs import error pops up, but the app runs fine otherwise. A lot of `ENOENT` for translation files are thrown in the `stderr` (also `aarch64-darwin`), but the language selector in the app works - and the sidebar icon is a generic Electron Icon titled `Electron`, so something is still off :/

BonusPlay reviewed (commented):
tested element-desktop on x86 and it works, but I'm also not very happy about the `yarn.nix` hack

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 381196`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>element-desktop</li>
    <li>element-web (element-web-unwrapped)</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/by-name/el/element-desktop/package.nix:
```diff
@@ -42,29 +42,34 @@ stdenv.mkDerivation (
-      patchShebangs node_modules/
+      mkdir -p node_modules/
+      cp -r $offlineCache/node_modules/* node_modules/
+      sed -i 's/matrix-gen-i18n/node node_modules\/matrix-web-i18n\/scripts\/gen-i18n.js/gi' package.json
```
We should use substituteInPlace with --replace-fail to notice when this no longer matches.

teutat3s committed:
element-desktop: use electron_34, follow upstream

version bump, released in v1.11.92

https://github.com/element-hq/element-desktop/commit/9ce980d2006e26f32300acdd78d1bac497a87e73

transcaffeine commented:
Happy to say that i was indeed able to drop the linter by executing the steps in [`element-desktop`s `i18n` script](https://github.com/element-hq/element-desktop/blob/v1.11.94/package.json#L21) one-by-one, though this might not be as portable as always calling `i18n`...

I also cherry-picked @teutat3s electron34 update from https://github.com/NixOS/nixpkgs/pull/387245. Package builds and works on my `amd64-linux` machine with translation still working correctly, will test on an `aarch64-darwin` later today.

transcaffeine commented on pkgs/by-name/el/element-desktop/package.nix:
```diff
@@ -42,29 +42,34 @@ stdenv.mkDerivation (
+      typescript
       makeWrapper
       jq
+      nodePackages_latest.prettier
```
Linter successfully removed, see comment below about this potentially making updates a bit harder/annoying if upstream ever changes their i18n building

teutat3s commented:
It seems that with https://github.com/NixOS/nixpkgs/pull/387319 the build process works without further changes : )

SuperSandro2000 commented:
I tried rebasing this locally but still hit the same git error as described in the comment.

teutat3s commented:
Ah, thanks. I tested with `1.11.95-rc.0`. Maybe let's just wait for that version to avoid the yarn.nix hack?

teutat3s commented:
Full diff to get the rc working:

```
diff --git i/pkgs/by-name/el/element-desktop/element-desktop-pin.nix w/pkgs/by-name/el/element-desktop/element-desktop-pin.nix
index ecdec1f89c8c..a33f526aef45 100644
--- i/pkgs/by-name/el/element-desktop/element-desktop-pin.nix
+++ w/pkgs/by-name/el/element-desktop/element-desktop-pin.nix
@@ -1,7 +1,7 @@
 {
-  "version" = "1.11.91";
+  "version" = "1.11.95-rc.0";
   "hashes" = {
-    "desktopSrcHash" = "sha256-nHA/j9V+vZUgY+eCCp/iO458GrXSkla+ruJbJjh9NJw=";
-    "desktopYarnHash" = "09k4kislf1zimbyn6m68bh2s3kpn2cs9h04wqli3j3qgkplcvxdv";
+    "desktopSrcHash" = "sha256-XvEUNKiSur9qS3uEU5o5ufXIj+9Aq2zi3nJpVQ6ZN5c=";
+    "desktopYarnHash" = "sha256-h5ZjpInF/qh/Cm46uzi1tTEUIbmHC/muV6X/FxGyFms=";
   };
 }
diff --git i/pkgs/by-name/el/element-desktop/keytar/default.nix w/pkgs/by-name/el/element-desktop/keytar/default.nix
index 80c1828c3caf..a6548b041c25 100644
--- i/pkgs/by-name/el/element-desktop/keytar/default.nix
+++ w/pkgs/by-name/el/element-desktop/keytar/default.nix
@@ -24,11 +24,11 @@ let

 in
 stdenv.mkDerivation rec {
-  pname = "keytar";
+  pname = "keytar-forked";
   inherit (pinData) version;

   src = fetchFromGitHub {
-    owner = "atom";
+    owner = "shiftkey";
     repo = "node-keytar";
     rev = "v${version}";
     hash = pinData.srcHash;
diff --git i/pkgs/by-name/el/element-desktop/keytar/pin.json w/pkgs/by-name/el/element-desktop/keytar/pin.json
index 626f5a62ea1d..5eba4a97a822 100644
--- i/pkgs/by-name/el/element-desktop/keytar/pin.json
+++ w/pkgs/by-name/el/element-desktop/keytar/pin.json
@@ -1,5 +1,5 @@
 {
-  "version": "7.9.0",
-  "srcHash": "sha256-Mnl0Im2hZJXJEtyXb5rgMntekkUAnOG2MN1bwfgh0eg=",
-  "npmHash": "sha256-ldfRWV+HXBdBYO2ZiGbVFSHV4/bMG43U7w+sJ4kpVUY="
+  "version": "7.10.0",
+  "srcHash": "sha256-G0IYBVB37ANnI/XEyzahs9DbTQLgUKmkDoGryRkra08=",
+  "npmHash": "sha256-bd4YK3AuD0N5Q9ijx1jqBRBL0WADhw1ZlaxspEctse4="
 }
diff --git i/pkgs/by-name/el/element-desktop/package.nix w/pkgs/by-name/el/element-desktop/package.nix
index 00626d26ecae..f6a8921c4d1c 100644
--- i/pkgs/by-name/el/element-desktop/package.nix
+++ w/pkgs/by-name/el/element-desktop/package.nix
@@ -67,8 +67,8 @@ stdenv.mkDerivation (
       yarn --offline run i18n
       yarn --offline run build:res

-      rm -rf node_modules/matrix-seshat node_modules/keytar
-      ${lib.optionalString useKeytar "ln -s ${keytar} node_modules/keytar"}
+      rm -rf node_modules/matrix-seshat node_modules/keytar-forked
+      ${lib.optionalString useKeytar "ln -s ${keytar} node_modules/keytar-forked"}
       ln -s $seshat node_modules/matrix-seshat

       runHook postBuild
diff --git i/pkgs/by-name/el/element-web-unwrapped/element-web-pin.nix w/pkgs/by-name/el/element-web-unwrapped/element-web-pin.nix
index 75bc7666939a..586f52884a7d 100644
--- i/pkgs/by-name/el/element-web-unwrapped/element-web-pin.nix
+++ w/pkgs/by-name/el/element-web-unwrapped/element-web-pin.nix
@@ -1,7 +1,7 @@
 {
-  "version" = "1.11.91";
+  "version" = "1.11.95-rc.0";
   "hashes" = {
-    "webSrcHash" = "sha256-kdjkmVkoJuV3SBFkVQr4IAi69mAs8V5i3qFOd66BP2s=";
-    "webYarnHash" = "sha256-in7qiGIXP+Ki820RB/uB2st2FIwrxjqYpdOmmLI6RSM=";
+    "webSrcHash" = "sha256-AMK3AZL6Ek6iE1ES71lTpdhexTzLSMIPpXayI1XtcqA=";
+    "webYarnHash" = "sha256-e9FwYNZEuqV3i2CGw3Ia3SuRkEGEJWWXF6RFNFZPkmQ=";
   };
 }
```

Andrew15-5 commented:
It appears that my keyboard emits some keystrokes way too close to each other, which results in incorrect order inserted into the text field in Element. It actually doesn't happen in the latest Vekstop and I thought that it's probably due to the fact that they use different Electron versions. So I started searching for a new PR for Element and I found one.

I see that there are some issues with the final patch, so I tried just updating the Electron version and hope that it won't break something:

```nix
  package =
    (pkgs.element-desktop.override {electron_33 = pkgs.electron;})
    .overrideAttrs (_: finalAttrs: {
      installPhase =
        finalAttrs.installPhase
        + ''wrapProgram $out/bin/element-desktop --set GTK_THEME Adwaita:dark'';
    });
```

And indeed this fixed my issue. So for this PR I can only leave a question of why is the latest `pkgs.electron` isn't used by default? Will it actually break something? I would've liked it if I can update nixpkgs and with it all electron apps are updated too, not just Vekstop, but Element as well. For example `pkgs.youtube-music` also uses the latest version.

teutat3s commented:
element-desktop [v1.11.95](https://github.com/element-hq/element-desktop/releases/tag/v1.11.95) is out.

transcaffeine committed:
element-desktop.keytar: 7.9.0 -> 7.10.0

Switch from atom/keytar to shiftkey/keytar-forked

transcaffeine commented:
With @teutat3s patch to keytar, i was able to successfully build and run `element-desktop@1.11.95`

teutat3s commented on pkgs/by-name/el/element-desktop/package.nix:
```diff
@@ -82,6 +83,7 @@ stdenv.mkDerivation (
       ln -s '${element-web}' "$out/share/element/webapp"
       cp -r '.' "$out/share/element/electron"
       cp -r './res/img' "$out/share/element"
+      chmod -R "a+w" "$out/share/element/electron/node_modules"
```
This should no longer be necessary.

teutat3s commented on pkgs/by-name/el/element-desktop/package.nix:
```diff
@@ -64,11 +64,12 @@ stdenv.mkDerivation (
       yarn --offline run build:ts
-      yarn --offline run i18n
+      node node_modules/matrix-web-i18n/scripts/gen-i18n.js
+      yarn --offline run i18n:sort
```
This should no longer be necessary.

teutat3s reviewed (commented):
Thank you for keeping this PR going. I have two comments/nits.

teutat3s reviewed (approved):
Tested working on my machine
